zerver/openapi/markdown_extension.py

```diff
@@ -485,5 +494,43 @@ def render_description(self, function: str) -> List[str]:
         return description
 
 
+class APITitlePreprocessor(Preprocessor):
+    def __init__(self, md: markdown.Markdown, config: Mapping[str, Any]) -> None:
+        super().__init__(md)
+        self.api_url = config["api_url"]
+
+    def run(self, lines: List[str]) -> List[str]:
+        done = False
+        while not done:
+            for line in lines:
+                loc = lines.index(line)
+                match = MACRO_REGEXP_TITLE.search(line)
+
+                if match:
+                    function = match.group(2)
+                    text = self.render_title(function)
+                    # The line that contains the directive to include the macro
+                    # may be preceded or followed by text or tags, in that case
+                    # we need to make sure that any preceding or following text
+                    # stays the same.
+                    line_split = MACRO_REGEXP_TITLE.split(line, maxsplit=0)
+                    preceding = line_split[0]
+                    following = line_split[-1]
+                    text = [preceding, *text, following]
+                    lines = lines[:loc] + text + lines[loc + 1 :]
+                    break
+            else:
+                done = True
+        return lines
+
+    def render_title(self, function: str) -> List[str]:
+        title: List[str] = []
+        path, method = function.rsplit(":", 1)
+        title_dict = "# " + get_openapi_summary(path, method)
+        title_dict = title_dict.replace("{{api_url}}", self.api_url)
```

Comment:
why the variable name `title_Dict`?

Reply:
It was indeed a misleading name. Changed it now.